internal/repository/verify.go

```diff
@@ -22,47 +22,61 @@ import (
 // another is to create a new RSL entry for the current state.
 var ErrRefStateDoesNotMatchRSL = errors.New("Git reference's current state does not match latest RSL entry") //nolint:stylecheck
 
-func (r *Repository) VerifyRef(ctx context.Context, target string, latestOnly bool) error {
+func (r *Repository) VerifyRef(ctx context.Context, localRefName, remoteRefName string, latestOnly bool) error {
 	var (
 		expectedTip gitinterface.Hash
 		err         error
 	)
 
 	slog.Debug("Identifying absolute reference path...")
-	target, err = r.r.AbsoluteReference(target)
+	localRefName, err = r.r.AbsoluteReference(localRefName)
 	if err != nil {
 		return err
 	}
 
-	slog.Debug(fmt.Sprintf("Verifying gittuf policies for '%s'", target))
+	if remoteRefName == "" {
+		remoteRefName = localRefName
+	} else {
+		// remote ref name is different
+		// **We must consider RSL entries that have remoteRefName rather than
+		// localRefName**
+		slog.Debug("Name of reference differs on remote, identifying absolute reference path...")
+		remoteRefName, err = r.r.AbsoluteReference(remoteRefName)
+		if err != nil {
+			return err
+		}
+	}
+
+	slog.Debug(fmt.Sprintf("Verifying gittuf policies for '%s'", remoteRefName))
 
 	if latestOnly {
-		expectedTip, err = policy.VerifyRef(ctx, r.r, target)
+		expectedTip, err = policy.VerifyRef(ctx, r.r, remoteRefName)
 	} else {
-		expectedTip, err = policy.VerifyRefFull(ctx, r.r, target)
+		expectedTip, err = policy.VerifyRefFull(ctx, r.r, remoteRefName)
 	}
 	if err != nil {
 		return err
 	}
 
+	// To verify the tip, we _must_ use the localRefName
 	slog.Debug("Verifying if tip of reference matches expected value from RSL...")
-	if err := r.verifyRefTip(target, expectedTip); err != nil {
+	if err := r.verifyRefTip(localRefName, expectedTip); err != nil {
```

Comment:
What's the difference between VerifyRef and verifyRefTip, and why is okay to replace the ref name for one and not the other?

(probably worth adding context here)

Reply:
verifyRefTip just checks the HEAD of the ref in the local repository to see that it matches the latest RSL entry for that ref. I can add the context as a comment.

Reply:
added a docstring for verifyRefTip.